pkg/client/unversioned/portforward/portforward.go

```diff
@@ -250,67 +254,79 @@ func (pf *PortForwarder) waitForConnection(listener net.Listener, port Forwarded
 func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	defer conn.Close()
 
-	glog.Infof("Handling connection for %d", port.Local)
+	pf.connectionIDLock.Lock()
+	connectionID := pf.nextConnectionID
+	pf.nextConnectionID++
```

Comment:
Make issuing the next id self-contained in a function that locks, defers an unlock, saves the current value, increments, and returns